File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/PaymentProduct5001SpecificOutputMapper.cs
using PaymentProduct5001SpecificOutputSdk = OnlinePayments.Sdk.Domain.PaymentProduct5001SpecificOutput;
using PaymentProduct5001SpecificOutputDto = Business.Domain.Payments.PaymentDetails.PaymentProduct5001SpecificOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class PaymentProduct5001SpecificOutputMapper
{
    public static PaymentProduct5001SpecificOutputDto Map(PaymentProduct5001SpecificOutputSdk? response)
    {
        return new()
        {
            Liability = response?.Liability,
            AccountNumber = response?.AccountNumber,
            AuthorisationCode = response?.AuthorisationCode,
            OperationCode = response?.OperationCode,
            MobilePhoneNumber = response?.MobilePhoneNumber
        };
    }
}

File: server/nodejs/src/infrastructure/mappers/payment-details/payment-product-5001-specific-output-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { PaymentProduct5001SpecificOutput } from '../../../business/domain/payments/payment-details/payment-product-5001-specific-output';

type PaymentProduct5001SpecificOutputSdk = Domain.PaymentProduct5001SpecificOutput;

export const PaymentProduct5001SpecificOutputMapper = {
    fromSdkResponse: (
        response?: PaymentProduct5001SpecificOutputSdk | null,
    ): PaymentProduct5001SpecificOutput | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            liability: response?.liability,
            accountNumber: response?.accountNumber,
            authorisationCode: response?.authorisationCode,
            operationCode: response?.operationCode,
            mobilePhoneNumber: response?.mobilePhoneNumber,
        };
    },
};
